### ISSUE TITLE
run  MrAdvice  not build

### ISSUE THREAD

**Issue by withlin** [ArxOne/MrAdvice#135]
i am run MrAdvice in dotnetcore version: 2.1.102 ,OS:windows10, It work, but i am use dotnet command: dotnet build .which is error.

The error Message:
```c#
Could not load file or assembly 'Microsoft.Build.Utilities.v4.0, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
```

**Comment by WithLin** [comment]
```C#
class Program
    {
        
        static void Main(string[] args)
        {
            Person p=new Person();
            p.Test();
            Console.WriteLine("Hello World!");
        }

    }

    public class Person
    {
        [MyProudAdvice]
        public void Test()
        {
            Console.WriteLine("........");
        }
    }

    public class MyProudAdvice : Attribute, IMethodAdvice
    {
        public void Advise(MethodAdviceContext context)
        {
           
            // do things you want here
            Console.WriteLine("===========");
            context.Proceed(); // this calls the original method
            // do other things here
            Console.WriteLine("===========");
        }
    }
```
so i am used dotnet core command ,which is error.
Can i help you?

**Comment by picrap** [comment]
It's not my fault 😢, this is related to the way dotnet is designed, and a lot of people bother with that, as you can see in https://github.com/Microsoft/msbuild/issues/2111

**Comment by WithLin** [comment]
emmmm,Thank you for your answer.
This is very embarrassing.
Is this the code you wrote and it conflicts with it?
If so, have you considered another way to achieve it?

**Comment by picrap** [comment]
I have no idea whether it can be done or not. The main problems are:
- It is based on `ITask` (it's a build task) which is part of MSBuild for .NET framework.
- It depends on other assemblies that also require to be migrated to .NET Core.
I'll take a quick look and let you know.

**Comment by WithLin** [comment]
hey guy, is there any progress?

**Comment by picrap** [comment]
Yes... Sort of: I broker most of the builds, so I need to fix all before new tests.
Don't expect anything too soon.

**Comment by WithLin** [comment]
I want to ask you some questions about this framework.
Does this framework support dynamic interception? It is similar to: giving the namespace + class name to the current framework , and then the framework helps me to intercept（AOP）.

**Comment by picrap** [comment]
I'm not sure I understand what you mean, but probably not: the aspects are injected at build-time, not at run-time.

**Comment by WithLin** [comment]
I want to inject it when I build it, but not in the Attribute way, because I want to inject some clients, such as rabbitmq, or redis dll inside the method, etc. These dlls, these dlls, I can not use the Attribute method, I want to pass the namespace and the classes and methods, then to the current framework, the framework helps me find him, then do before and after

**Comment by picrap** [comment]
MrAdvice does not do this. However, that's a good idea and could be a great feature.

**Comment by WithLin** [comment]
Google Translate translates Chinese into English. which it is  a lot worse.

**Comment by WithLin** [comment]
Because I want to do something in distributed tracing with google Dapper paper,
Tracking components such as rabbitmq，redis，mvc，grpc，httpclient, which is minimal for code intrusion, is a scenario.

**Comment by WithLin** [comment]
Do you have an app like WeChat?
I can add you as my friend.
hei hei

**Comment by picrap** [comment]
I use Trillian (account: picrap) and have an XMPP account (pascal@craponne.fr) or Google Talk (picrap@gmail.com)

**Comment by WithLin** [comment]
hi ,  i am fix it .

```c#
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="MrAdvice" Version="2.5.25" ExcludeAssets="build" />
  </ItemGroup>

</Project>
```
add ExcludeAssets="build"  in  PackageReference  ,which is ok.

**Comment by WithLin** [comment]
if you have other idea, please tell me .Thank you.

**Comment by picrap** [comment]
Good! I'll see if I can include your fix in the package.

**Comment by WithLin** [comment]
i found a problem. i have a name of test class library reference MrAdvice dll,but  the console library  reference  test class library.but aop.it does not work. so i try reference MrAdvice in console library.so it work. why ? so  i want to down two MrAdvice dll ,Which is redundancy.

**Comment by WithLin** [comment]
cure the symptoms, not the disease. i think that we should fix MrAdvice code problem.

**Comment by picrap** [comment]
Regarding your so-called problem, your console application must also use MrAdvice (and not only reference the dll), otherwise, it won't be weaved at build.

**Comment by WithLin** [comment]
yeah.so i add it. but add ExcludeAssets="build" in PackageReference ,which does not work.

**Comment by picrap** [comment]
Fixed in 2.9
